proposals/0033-data-collections.md

```diff
@@ -0,0 +1,244 @@
+- Start Date: 2023-05-01
+- Reference Issues: https://github.com/withastro/roadmap/issues/530
+- Implementation PR: https://github.com/withastro/astro/pull/6850
+
+# Summary
+
+Introduce a standard to store data separately from your content (ex. JSON files).
+
+# Example
+
+Like content collections, data collections are created in the `src/content/` directory. These collections should include only JSON or YAML files:
+
+```
+src/content/
+  authors/
+    ben.json
+    tony.json
+```
+
+These collections are configured using the same `defineCollection()` utility in your `content/config.ts` with `type: 'data'` specified:
+
+```ts
+// src/content/config.ts
+import { defineCollection, z } from 'astro:content';
+
+const authors = defineCollection({
+  type: 'data',
+  schema: z.object({
+    name: z.string(),
+    twitter: z.string().url(),
+  })
+});
+
+export const collections = { authors };
+```
+
+These collections can also be queried using the `getCollection()` and `getEntry()` utilities:
+
+```astro
+---
+import { getCollection, getEntry } from 'astro:content';
+
+const authors = await getCollection('authors');
+const ben = await getEntry('authors', 'ben');
+---
+
+<p>
+  Authors: {authors.map(author => author.data.name).join(', ')}
+</p>
+<p>
+  The coolest author: <a href={ben.data.twitter}>{ben.data.name}</a>
+</p>
+```
+
+# Background & Motivation
+
+Content collections are restricted to supporting `.md`, `.mdx`, and `.mdoc` files. This is limiting for other forms of data you may need to store, namely raw data formats like JSON or YAML.
+
+Taking a blog post as the example, there will likely be author information thats reused across multiple blog posts. To standardize updates when, say, updating an author's profile picture, it's best to store authors in a standalone data entry.
+
+The content collections API was built generically to support this future, choosing format-agnostic naming like `data` instead of `frontmatter` and `body` instead of `rawContent`. Because of this, expanding support to new data formats without API changes is a natural progression.
+
+# Goals
+
+- **Introduce JSON and YAML collection support,** configurable and queryable with similar APIs to content collections.
+- **Determine where data collections are stored.** We may allow data collections within `src/content/`, or introduce a new reserved directory.
```

Comment:
Should we update this line and just say that `src/content/` directory has been chosen?

Reply:
Probably... could remove the goal entirely?

Reply:
Yeah I think so :)